socket_mock_wrap_test/src/ux_socket.cpp

```diff
@@ -0,0 +1,81 @@
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+#include <iostream>
+
+#include "ux_socket.hpp"
+#include "ux_socket_impl.hpp"
+
+inline void exit_error(char const* msg)
+{
+    if (msg != nullptr)
+        std::cerr << msg << '\n';
+    exit(EXIT_FAILURE);
+}
+
+UxSocket::UxSocket(UxSocketInterface* isock_impl) 
+    : sock_impl_(isock_impl)
+{
+    sock_fd_ = 
+            sock_impl_->socket(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC | SOCK_NONBLOCK, 0);
```

Comment:
If I'm not wrong, we should use AF_INET sockets, to allow for "communication between multiple ECUs in the future". AF_UNIX is limited to one ECU